Tools/machines/lxplus-cern/spack.yaml

```
@@ -0,0 +1,47 @@
+spack:
+# CUDA
+  definitions:
+  - cuda: []
+  - cuda: [cuda ^ncurses+symlinks, blaspp +cuda ^ncurses+symlinks]
+    when: env.get("SPACK_STACK_USE_CUDA", "") == "1"
+
+# basic Python dependencies for WarpX Python PICMI bindings
+  - python: []
+  - python: [python ^ncurses+symlinks, py-pip ^ncurses+symlinks, py-mpi4py ^ncurses+symlinks, py-numpy ^ncurses+symlinks,
+      py-periodictable ^ncurses+symlinks, py-picmistandard ^ncurses+symlinks]
+    when: env.get("SPACK_STACK_USE_PYTHON", "") == "1"
+  specs:
+  - ncurses+symlinks
+  - adios2 ^ncurses+symlinks
+  - blaspp ^ncurses+symlinks
+  - ccache ^ncurses+symlinks
+  - cmake ^ncurses+symlinks
+  - fftw ^ncurses+symlinks
+  - hdf5 ^ncurses+symlinks
+  - lapackpp ^ncurses+symlinks
+  - mpi ^ncurses+symlinks
+  - openpmd-api ^ncurses+symlinks
+  - pkgconfig
+  - $cuda
+  - $python
+
+  packages:
+    all:
+      target: [x86_64]
+      compiler: [gcc@9.2.0]
```

We can save compile time by compiling with `~fortran` for all packages (Spack defaults to `+fortran` for packages that have this variant).

And we can narrow down the virtual `mpi` package to only use `OpenMPI`:
```suggestion
      compiler: [gcc@9.2.0]
      variants: ~fortran
      providers:
        mpi: [openmpi]
```

---

Nice! I wasn't aware of these tricks!